flaky test TestSyncTimeout

Seen in https://travis-ci.org/coreos/etcd/builds/39440213:

```
Running tests...
ok      github.com/coreos/etcd/client   1.071s  coverage: 46.9% of statements
ok      github.com/coreos/etcd/discovery    3.198s  coverage: 71.7% of statements
ok      github.com/coreos/etcd/etcdctl/command  1.026s  coverage: 1.9% of statements
2014/10/29 23:18:24 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:24 raft: leader changed from 0 to 1
2014/10/29 23:18:24 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:24 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:24 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:24 raft: leader changed from 0 to 1
2014/10/29 23:18:24 raft: leader changed from 0 to 1
2014/10/29 23:18:24 raft: leader changed from 0 to 1
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 raft: leader changed from 0 to bad0
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 raft: leader changed from 0 to bad0
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 raft: leader changed from 0 to bad0
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 raft: leader changed from 0 to bad0
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
--- FAIL: TestSyncTimeout (0.01 seconds)
    server_test.go:715: CallSyncTime = 5.781003ms, want < 1ms
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 raft: leader changed from 0 to bad0
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 etcdserver: set snapshot count to default 10000
2014/10/29 23:18:25 etcdserver: published {Name:node1 ClientURLs:[http://a http://b]} to cluster 0
2014/10/29 23:18:25 etcdserver: aborting publish because server is stopped
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: publish error: etcdserver: request timed out
2014/10/29 23:18:25 etcdserver: aborting publish because server is stopped
FAIL
coverage: 62.7% of statements
FAIL    github.com/coreos/etcd/etcdserver   0.510s
ok      github.com/coreos/etcd/etcdserver/etcdhttp  1.115s  coverage: 93.3% of statements
ok      github.com/coreos/etcd/etcdserver/etcdhttp/httptypes    1.153s  coverage: 77.4% of statements
?       github.com/coreos/etcd/etcdserver/etcdserverpb  [no test files]
ok      github.com/coreos/etcd/integration  2.672s  coverage: 0.0% of statements
ok      github.com/coreos/etcd/pkg/flags    1.022s  coverage: 75.0% of statements
ok      github.com/coreos/etcd/pkg/strutil  1.023s  coverage: 100.0% of statements
ok      github.com/coreos/etcd/pkg/transport    1.049s  coverage: 69.7% of statements
ok      github.com/coreos/etcd/proxy    1.084s  coverage: 87.8% of statements
ok      github.com/coreos/etcd/raft 3.384s  coverage: 91.5% of statements
ok      github.com/coreos/etcd/snap 1.026s  coverage: 80.6% of statements
ok      github.com/coreos/etcd/store    1.275s  coverage: 89.2% of statements
ok      github.com/coreos/etcd/wait 1.020s  coverage: 100.0% of statements
ok      github.com/coreos/etcd/wal  1.605s  coverage: 80.1% of statements
ok      github.com/coreos/etcd  1.024s  coverage: 21.1% of statements

```

--- a/etcdserver/server_test.go
+++ b/etcdserver/server_test.go
@@ -678,12 +678,17 @@ func TestSync(t *testing.T) {
 	srv := &EtcdServer{
 		node: n,
 	}
-	start := time.Now()
-	srv.sync(defaultSyncTimeout)
+	done := make(chan struct{})
+	go func() {
+		srv.sync(10 * time.Second)
+		close(done)
+	}()
 
 	// check that sync is non-blocking
-	if d := time.Since(start); d > time.Millisecond {
-		t.Errorf("CallSyncTime = %v, want < %v", d, time.Millisecond)
+	select {
+	case <-done:
+	case <-time.After(time.Second):
+		t.Fatalf("sync should be non-blocking but did not return after 1s!")
 	}
 
 	testutil.ForceGosched()
@@ -707,12 +712,17 @@ func TestSyncTimeout(t *testing.T) {
 	srv := &EtcdServer{
 		node: n,
 	}
-	start := time.Now()
-	srv.sync(0)
+	done := make(chan struct{})
+	go func() {
+		srv.sync(0)
+		close(done)
+	}()
 
 	// check that sync is non-blocking
-	if d := time.Since(start); d > time.Millisecond {
-		t.Errorf("CallSyncTime = %v, want < %v", d, time.Millisecond)
+	select {
+	case <-done:
+	case <-time.After(time.Second):
+		t.Fatalf("sync should be non-blocking but did not return after 1s!")
 	}
 
 	// give time for goroutine in sync to cancel

